app/helpers/admin/injection_helper.rb

```diff
@@ -34,13 +34,6 @@ def admin_inject_enterprise_roles
                                   Api::Admin::EnterpriseRoleSerializer
     end
 
-    def admin_inject_payment_methods
-      admin_inject_json_ams_array "admin.paymentMethods",
```

Comment:
I can see that AngularJS module `admin.enterprises` use `admin.paymentMethods` defined in `app/assets/javascripts/admin/enterprises/enterprises.js.coffee`: 

```
angular.module("admin.enterprises", [
  "admin.paymentMethods",
...
```

But it turns out that `app/assets/javascripts/admin/enterprises/services/enterprise_payment_methods.js.coffee` use `PaymentMethods` as seen: 
```
 angular.module("admin.enterprises")
  .factory "EnterprisePaymentMethods", (enterprise, PaymentMethods) ->
```

I suppose this comes from what you've deleted. Did you. test that?

Reply:
I'm reading the code in `app/assets/javascripts/admin/enterprises/services/enterprise_payment_methods.js.coffee`

```coffee
angular.module("admin.enterprises")
  .factory "EnterprisePaymentMethods", (enterprise, PaymentMethods) ->
    new class EnterprisePaymentMethods
      paymentMethods: PaymentMethods.all
      console.log(PaymentMethods)

      constructor: ->
        for payment_method in @paymentMethods
          payment_method.selected = payment_method.id in enterprise.payment_method_ids

      displayColor: ->
        if @paymentMethods.length > 0 && @selectedCount() > 0
          "blue"
        else
          "red"

      selectedCount: ->
        @paymentMethods.reduce (count, payment_method) ->
          count++ if payment_method.selected
          count
        , 0
```

I think it's job is to append (a css class) "blue" or "red" somewhere. 

```coffee
displayColor: ->
        if @paymentMethods.length > 0 && @selectedCount() > 0
          "blue"
        else
          "red"
```

But I can't figure out where. It's in this page for sure (or the equivalent page for the enterprise manager, I'm logged in as super here) I checked using a couple of `console.logs` and it only logs when I'm in this page.

![payment](https://user-images.githubusercontent.com/87677429/183049906-6e269996-68c6-4be2-a771-ac37718f9ba0.png)

One odd thing I noticed is that when I `console.log(PaymentMethods)` inside this file it's returning an empty object (I checked this in `master`). Maybe it was used for something else before and now is redundant? Can you think of anything inside the `Payment Methods` tab that turns red after an input change?